### ISSUE TITLE
bumblebee applet

### ISSUE THREAD

**Issue by intervisionlord** [pdcurtis/cinnamon-applets#5]
## Issue Description
When applet is started, it use GPU so it always on and never turns back to intel graphics

### OS
Debian 10.1 (x86-64)

### Kernel
4.19.0-6-amd64

### Cinnamon
3.8.8

### nVidia Driver
nvidia-legacy-390xx-driver

### Bumblebee Version
3.2.1

### Applet Version
3.2.5

## How to reproduce the problem

1. Add applet to the panel
2. It shows that NVIDIA card is used now and shows it's temperature
3. You may also run `watch -n 1 'ps aux | grep -i optirun'` to see that applet runs optirun periodicaly

You may also check the syslog file and see that every second driver loads and unloads

```
> tail /var/log/messages 
Sep  8 13:28:56 twrnb kernel: [  741.238322] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  390.116  Sun Jan 27 06:30:32 PST 2019
Sep  8 13:28:57 twrnb kernel: [  742.176814] nvidia-modeset: Unloading
Sep  8 13:28:58 twrnb kernel: [  743.255335] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  390.116  Sun Jan 27 06:30:32 PST 2019
Sep  8 13:28:59 twrnb kernel: [  744.263520] nvidia-modeset: Unloading
Sep  8 13:29:00 twrnb kernel: [  745.243119] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  390.116  Sun Jan 27 06:30:32 PST 2019
Sep  8 13:29:01 twrnb kernel: [  746.179262] nvidia-modeset: Unloading
Sep  8 13:29:02 twrnb kernel: [  747.246462] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  390.116  Sun Jan 27 06:30:32 PST 2019
Sep  8 13:29:03 twrnb kernel: [  748.213356] nvidia-modeset: Unloading
Sep  8 13:29:04 twrnb kernel: [  749.267709] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  390.116  Sun Jan 27 06:30:32 PST 2019
Sep  8 13:29:05 twrnb kernel: [  750.217331] nvidia-modeset: Unloading

```

**Comment by pdcurtis** [comment]
@intervisionlord I have been unable to reproduce this with the same driver, applet and bumblebee versions under Mint 19.2 with Cinnamon 4.2.3 and a 4.15.60-generic kernel.
Does bumblebee work correctly without the applet loaded and after the applet has been unloaded?
Is there any other switching mechanism loaded like nvidia-prime?
Have you looked at `dmesg | grep -i bb` and `dmesg | grep -i nvidia` in particular after a reboot ?

**Comment by intervisionlord** [comment]
> Does bumblebee work correctly without the applet loaded and after the applet has been unloaded?

Yes, it works perfect

> Is there any other switching mechanism loaded like nvidia-prime?

No, only bbswitch

> Have you looked at dmesg | grep -i bb and dmesg | grep -i nvidia in particular after a reboot ?

I can check it right now (after reboot)

```
> dmesg | grep -i bb 
[    2.203237] Loaded X.509 cert 'Debian Secure Boot CA: 6ccece7e4c6c0d1f6149f3dd27dfcc5cbb419ea1'
[    6.683740] bbswitch: loading out-of-tree module taints kernel.
[    6.683856] bbswitch: module verification failed: signature and/or required key missing - tainting kernel
[    6.684163] bbswitch: version 0.8
[    6.684172] bbswitch: Found integrated VGA device 0000:00:02.0: \_SB_.PCI0.GFX0
[    6.684179] bbswitch: Found discrete VGA device 0000:01:00.0: \_SB_.PCI0.RP01.PXSX
[    6.684387] bbswitch: detected an Optimus _DSM function
[    6.684481] bbswitch: Succesfully loaded. Discrete card 0000:01:00.0 is on
[    6.690957] bbswitch: disabling discrete graphics
```

```
> dmesg | grep -i nvidia 
[  737.231227] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  390.116  Sun Jan 27 06:30:32 PST 2019
[  738.158626] nvidia-modeset: Unloading
[  739.250118] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  390.116  Sun Jan 27 06:30:32 PST 2019
[  740.185893] nvidia-modeset: Unloading
[  741.238322] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  390.116  Sun Jan 27 06:30:32 PST 2019
[  742.176814] nvidia-modeset: Unloading
[  743.255335] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  390.116  Sun Jan 27 06:30:32 PST 2019
[  744.263520] nvidia-modeset: Unloading
[  745.243119] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  390.116  Sun Jan 27 06:30:32 PST 2019
[  746.179262] nvidia-modeset: Unloading
[  747.246462] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  390.116  Sun Jan 27 06:30:32 PST 2019
[  748.213356] nvidia-modeset: Unloading
[  749.267709] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  390.116  Sun Jan 27 06:30:32 PST 2019
[  750.217331] nvidia-modeset: Unloading
```

Strange timestamp O_o

Oh...

```
 > optirun --status
Bumblebee status: Ready (3.2.1). X inactive. Discrete video card is on.
```

Looks like a problem with drivers or bumblebee?

**Comment by intervisionlord** [comment]
So, it seems that i have found a solution.

There was no acpi-call installed on system, so

1. `> apt-get install acpi-call-dkms`
2. reboot
3. run `optirun --status` to check
4. `Bumblebee status: Ready (3.2.1). X inactive. Discrete video card is off. `
5. Launch the applet
6. It shows that intel graphics is in use
7. Launch some program with optirun
8. Applet shows that nVidia graphics used now
9. run `optirun --status` again to check
10. `Bumblebee status: Ready (3.2.1). X is PID 3177, 1 applications using bumblebeed.`

Nice...

So, thank you for participation and sry for this issue )

**Comment by pdcurtis** [comment]
@intervisionlord No problem. Glad you have sorted it out.

Is this going to be a problem with all Debian systems?

**Comment by intervisionlord** [comment]
Actualy i don't know, but it seems so (recently released debian 10 and 10.1 have tonns of strange issues according to personal experience and messages on forums about it)
